From dcbb2d2c3a3d4f22f91bbbc006989338fd556ab9 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Fri, 2 Aug 2019 08:00:10 +0900
Subject: [PATCH] i915: ipts: add destroy_doorbell() when disabling ipts guc
 submission

Adding destroy_doorbell() into i915_guc_ipts_submission_disable()
corresponds to __guc_client_disable() on 5.0+
---
 drivers/gpu/drm/i915/intel_guc_submission.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_guc_submission.c b/drivers/gpu/drm/i915/intel_guc_submission.c
index 32a9d29c5..4cbc75fe3 100644
--- a/drivers/gpu/drm/i915/intel_guc_submission.c
+++ b/drivers/gpu/drm/i915/intel_guc_submission.c
@@ -1408,6 +1408,7 @@ void i915_guc_ipts_submission_disable(struct drm_i915_private *dev_priv)
 	if (!guc->ipts_client)
 		return;
 
+	destroy_doorbell(guc->ipts_client);
 	guc_client_free(guc->ipts_client);
 	guc->ipts_client = NULL;
 }
-- 
2.22.0

